### DISCUSSION TITLE
"Newbie" Can I bring this EFE / Engelman Sensostar U Heat Meter device to run with wmbusmeters?

### DISCUSSION THREAD

**Discussion by addy38** [wmbusmeters/wmbusmeters#666] [Q&A] [answered]
Sorry, to write here but I have no idea where else I could ask. 
After 1 week reading and testing I'am lost.
Hardware:
- SensoStar U Energy Meter with wired MBus board
- DYKB USB to MBUS Master
 - Rasberry 2B
 - at the End the Energy Meter values should go via MQTT to my Homematic System 

I am no really familar with Raspberry but I think I could install at the Raspberry the wmbusmeter like
sudo apt update
sudo apt install snapd
sudo reboot
sudo snap install core
wmbusmeters
sudo snap install wmbusmeters

now I can find the files in /snapt/wmbusmeters but I'am not able to start the software or do any scan to see if the System detect my Meter 
Can someone give me some hints how to go on?

**Comment by addy38** [comment]
Hi, after 1 more day I have libmbus running and got following readout. Can I setup wmbusmeters to work with it?
` /usr/local/bin/mbus-serial-request-data -b 2400 /dev/ttyUSB0 0
<?xml version="1.0" encoding="ISO-8859-1"?>
<MBusData>

    <SlaveInformation>
        <Id>23556896</Id>
        <Manufacturer>EFE</Manufacturer>
        <Version>0</Version>
        <ProductName>Engelmann / Elster SensoStar 2</ProductName>
        <Medium>Heat: Outlet</Medium>
        <AccessNumber>190</AccessNumber>
        <Status>00</Status>
        <Signature>0000</Signature>
    </SlaveInformation>

    <DataRecord id="0">
        <Function>Instantaneous value</Function>
        <StorageNumber>0</StorageNumber>
        <Unit>Fabrication number</Unit>
        <Value>23556896</Value>
        <Timestamp>2022-11-13T17:41:40Z</Timestamp>
    </DataRecord>

    <DataRecord id="1">
        <Function>Instantaneous value</Function>
        <StorageNumber>0</StorageNumber>
        <Unit>Energy (kWh)</Unit>
        <Value>55</Value>
        <Timestamp>2022-11-13T17:41:40Z</Timestamp>
    </DataRecord>

    <DataRecord id="2">
        <Function>Instantaneous value</Function>
        <StorageNumber>0</StorageNumber>
        <Unit>Volume (m m^3)</Unit>
        <Value>13544</Value>
        <Timestamp>2022-11-13T17:41:40Z</Timestamp>
    </DataRecord>

    <DataRecord id="3">
        <Function>Instantaneous value</Function>
        <StorageNumber>0</StorageNumber>
        <Unit>Power (W)</Unit>
        <Value>0</Value>
        <Timestamp>2022-11-13T17:41:40Z</Timestamp>
    </DataRecord>

    <DataRecord id="4">
        <Function>Maximum value</Function>
        <StorageNumber>0</StorageNumber>
        <Unit>Power (W)</Unit>
        <Value>277537</Value>
        <Timestamp>2022-11-13T17:41:40Z</Timestamp>
    </DataRecord>

    <DataRecord id="5">
        <Function>Instantaneous value</Function>
        <StorageNumber>0</StorageNumber>
        <Unit>Volume flow (m m^3/h)</Unit>
        <Value>0</Value>
        <Timestamp>2022-11-13T17:41:40Z</Timestamp>
    </DataRecord>

    <DataRecord id="6">
        <Function>Maximum value</Function>
        <StorageNumber>0</StorageNumber>
        <Unit>Volume flow (m m^3/h)</Unit>
        <Value>6550</Value>
        <Timestamp>2022-11-13T17:41:40Z</Timestamp>
    </DataRecord>

    <DataRecord id="7">
        <Function>Instantaneous value</Function>
        <StorageNumber>0</StorageNumber>
        <Unit>Flow temperature (deg C)</Unit>
        <Value>35</Value>
        <Timestamp>2022-11-13T17:41:40Z</Timestamp>
    </DataRecord>

    <DataRecord id="8">
        <Function>Instantaneous value</Function>
        <StorageNumber>0</StorageNumber>
        <Unit>Return temperature (deg C)</Unit>
        <Value>31</Value>
        <Timestamp>2022-11-13T17:41:40Z</Timestamp>
    </DataRecord>

    <DataRecord id="9">
        <Function>Instantaneous value</Function>
        <StorageNumber>0</StorageNumber>
        <Unit>Temperature Difference (1e-2  deg C)</Unit>
        <Value>427</Value>
        <Timestamp>2022-11-13T17:41:40Z</Timestamp>
    </DataRecord>

    <DataRecord id="10">
        <Function>Instantaneous value</Function>
        <StorageNumber>0</StorageNumber>
        <Unit>On time (days)</Unit>
        <Value>68</Value>
        <Timestamp>2022-11-13T17:41:40Z</Timestamp>
    </DataRecord>

    <DataRecord id="11">
        <Function>Instantaneous value</Function>
        <StorageNumber>0</StorageNumber>
        <Unit>Time Point (time &amp; date)</Unit>
        <Value>2022-11-13T18:41:00</Value>
        <Timestamp>2022-11-13T17:41:40Z</Timestamp>
    </DataRecord>

    <DataRecord id="12">
        <Function>Instantaneous value</Function>
        <StorageNumber>1</StorageNumber>
        <Unit>Energy (kWh)</Unit>
        <Value>0</Value>
        <Timestamp>2022-11-13T17:41:40Z</Timestamp>
    </DataRecord>

    <DataRecord id="13">
        <Function>Instantaneous value</Function>
        <StorageNumber>1</StorageNumber>
        <Unit>Volume (m m^3)</Unit>
        <Value>0</Value>
        <Timestamp>2022-11-13T17:41:40Z</Timestamp>
    </DataRecord>

    <DataRecord id="14">
        <Function>Instantaneous value</Function>
        <StorageNumber>1</StorageNumber>
        <Unit>Time Point (date)</Unit>
        <Value>2000-00-00</Value>
        <Timestamp>2022-11-13T17:41:40Z</Timestamp>
    </DataRecord>

    <DataRecord id="15">
        <Function>Instantaneous value</Function>
        <StorageNumber>0</StorageNumber>
        <Unit>Error flags</Unit>
        <Value>0</Value>
        <Timestamp>2022-11-13T17:41:40Z</Timestamp>
    </DataRecord>

    <DataRecord id="16">
        <Function>Instantaneous value</Function>
        <StorageNumber>0</StorageNumber>
        <Unit>Model / Version</Unit>
        <Value>5</Value>
        <Timestamp>2022-11-13T17:41:40Z</Timestamp>
    </DataRecord>

    <DataRecord id="17">
        <Function>Instantaneous value</Function>
        <StorageNumber>0</StorageNumber>
        <Tariff>2</Tariff>
        <Device>0</Device>
        <Unit>Energy (kWh)</Unit>
        <Value>0</Value>
        <Timestamp>2022-11-13T17:41:40Z</Timestamp>
    </DataRecord>

    <DataRecord id="18">
        <Function>Instantaneous value</Function>
        <StorageNumber>1</StorageNumber>
        <Tariff>2</Tariff>
        <Device>0</Device>
        <Unit>Energy (kWh)</Unit>
        <Value>0</Value>
        <Timestamp>2022-11-13T17:41:40Z</Timestamp>
    </DataRecord>

    <DataRecord id="19">
        <Function>Instantaneous value</Function>
        <StorageNumber>0</StorageNumber>
        <Tariff>3</Tariff>
        <Device>0</Device>
        <Unit>Energy (kWh)</Unit>
        <Value>0</Value>
        <Timestamp>2022-11-13T17:41:40Z</Timestamp>
    </DataRecord>

    <DataRecord id="20">
        <Function>Instantaneous value</Function>
        <StorageNumber>1</StorageNumber>
        <Tariff>3</Tariff>
        <Device>0</Device>
        <Unit>Energy (kWh)</Unit>
        <Value>0</Value>
        <Timestamp>2022-11-13T17:41:40Z</Timestamp>
    </DataRecord>

    <DataRecord id="21">
        <Function>Instantaneous value</Function>
        <StorageNumber>0</StorageNumber>
        <Tariff>0</Tariff>
        <Device>1</Device>
        <Unit>Volume (m m^3)</Unit>
        <Value>0</Value>
        <Timestamp>2022-11-13T17:41:40Z</Timestamp>
    </DataRecord>

    <DataRecord id="22">
        <Function>Instantaneous value</Function>
        <StorageNumber>1</StorageNumber>
        <Tariff>0</Tariff>
        <Device>1</Device>
        <Unit>Volume (m m^3)</Unit>
        <Value>0</Value>
        <Timestamp>2022-11-13T17:41:40Z</Timestamp>
    </DataRecord>

    <DataRecord id="23">
        <Function>Instantaneous value</Function>
        <StorageNumber>0</StorageNumber>
        <Tariff>0</Tariff>
        <Device>2</Device>
        <Unit>Volume (m m^3)</Unit>
        <Value>0</Value>
        <Timestamp>2022-11-13T17:41:40Z</Timestamp>
    </DataRecord>

    <DataRecord id="24">
        <Function>Instantaneous value</Function>
        <StorageNumber>1</StorageNumber>
        <Tariff>0</Tariff>
        <Device>2</Device>
        <Unit>Volume (m m^3)</Unit>
        <Value>0</Value>
        <Timestamp>2022-11-13T17:41:40Z</Timestamp>
    </DataRecord>

    <DataRecord id="25">
        <Function>Instantaneous value</Function>
        <StorageNumber>0</StorageNumber>
        <Tariff>0</Tariff>
        <Device>3</Device>
        <Unit>Volume (m m^3)</Unit>
        <Value>0</Value>
        <Timestamp>2022-11-13T17:41:40Z</Timestamp>
    </DataRecord>

    <DataRecord id="26">
        <Function>Instantaneous value</Function>
        <StorageNumber>1</StorageNumber>
        <Tariff>0</Tariff>
        <Device>3</Device>
        <Unit>Volume (m m^3)</Unit>
        <Value>0</Value>
        <Timestamp>2022-11-13T17:41:40Z</Timestamp>
    </DataRecord>
`

**Comment by addy38** [comment]
@weetmuts 
sorry to Tag you but can you give me a hint how to go on with my request

**Comment by addy38** [comment, marked as answer]
Hi, just to end this and maybe someone else can use it.
At the end it is working, my config
`
loglevel=normal
device=MAIN=/dev/ttyUSB0:mbus:2400
logtelegrams=true
format=json
meterfiles=/var/lib/wmbusmeters/meter_readings
meterfilesaction=overwrite
meterfilesnaming=name
meterfilestimestamp=day
logfile=/var/log/wmbusmeters/wmbusmeters.log
shell=/usr/bin/mosquitto_pub -h MQTT_BROKER_IP -p 1883 -u MQTT_USERNAME -P MQTT_PASSWORD -t MQTT_TOPIC  -m "$METER_JSON"`

my Meter file, for the Engelmann Sensostar 2 Heat Meter I use the "microclima" driver, the most common values are present. 
`name=WMZ
id=23556896
key=NOKEY
pollinterval=120s
driver=microclima:MAIN:mbus`

**Reply by weetmuts** [reply]
Many thanks for the answer! Sorry about not answering! I am mostly looking at the issues, so I missed this question.
Please let me know if the README can be improced!

**Reply by Mufika** [reply]
@addy38 
Were you able to give commands using snaps?  Or did you just work with the config file. I'm completely lost.  When typing wmbusmeters it says command not found.   I would like to issue the command wmbusmeters --listmeters.  or is the snap only for running a service ?
